### PR TITLE
added an email option to accession upload

### LINKED ISSUE #4729: Implement email notifications for uploading large number of trials into the Database.
Expected Behavior 
--------------------------------------------------------------------------

Currently, we don't get an email notification about the status of uploading large numbers of trials using GUI.

For Bugs:
---------

### PR DESCRIPTION
**Pull request by Sri-2023**
Description 
-----------------------------------------------------

Also, will resolve #4729

Checklist 
---------------------------------------------------------------------------------------
- [ ] Refactoring only
- [ ] Documentation only
- [ ] Fixture update only
- [ ] Bug fix
  - [ ] The relevant issue has been closed.
  - [ ] Further work is required.
- [ ] New feature
  - [ ] Relevant tests have been created and run.
  - [ ] Data was added to the fixture
    - [ ] Data was added via a patch in `/t/data/fixture/patches/`.
  - [ ] User-Facing Change
    - [ ] The user manual in `/docs` has been updated.
  - [ ] Any new Perl has been documented using **perldoc**.
  - [ ] Any new JavaScript has been documented using **JSDoc**.
  - [ ] Any new _legacy_ JavaScript has been moved from `/js` to `/js/source/legacy`.

### CODE REVIEW

**Comment by Sri-2023** [comment]
@dwaring87, thank you for testing and providing feedback. I will fix the issue soon. Earlier, was able to upload accessions being on the master. However, after performing a git pull, am unable to upload accessions. I'm encountering a long error message,
Screenshot from 2024-07-12 13-19-05

**Comment by Sri-2023** [comment]
Thank you @dwaring87 for fixing the bug. It works now..

**Review by dwaring87 — APPROVED** [review]
@Sri-2023 Thanks for the updates!

File: mason/breeders_toolbox/manage_accessions.mas [hunk 1]
```diff
@@ -6,6 +6,7 @@ $list_id => undef
 $editable_stock_props => {}
 $editable_stock_props_definitions => {}
 $show_grafting_interface => ''
+$email_address => ""
```
Comment (dwaring87) [inline mason/breeders_toolbox/manage_accessions.mas:9]: this is not needed

File: mason/breeders_toolbox/trialtreefolders.mas [hunk 2]
```diff
@@ -8,6 +8,7 @@ $editable_stock_props_definitions => {}
 $timestamp
 $management_factor_types => ()
 $design_types => ()
+$email_address => ""
```
Comment (dwaring87) [inline mason/breeders_toolbox/trialtreefolders.mas:11]: this is not needed

File: mason/breeders_toolbox/trialtreefolders.mas [hunk 3]
```diff
@@ -24,7 +25,7 @@ $design_types => ()
 <& /breeders_toolbox/trial/download_trials_phenotypes_dialog.mas &>
 
 <& /breeders_toolbox/trial/trial_upload_dialogs.mas, locations => \@locations, breeding_programs => $breeding_programs, design_types => $design_types &>
-<& /breeders_toolbox/add_accessions_dialogs.mas, preferred_species=>$preferred_species, editable_stock_props=>$editable_stock_props, editable_stock_props_definitions=>$editable_stock_props_definitions &>
+<& /breeders_toolbox/add_accessions_dialogs.mas, preferred_species=>$preferred_species, editable_stock_props=>$editable_stock_props, editable_stock_props_definitions=>$editable_stock_props_definitions, email_address => $email_address &>
```
Comment (dwaring87) [inline mason/breeders_toolbox/trialtreefolders.mas:28]: email_address argument is not needed

**Review by dwaring87 — APPROVED** [review]
👍🏽

**Comment by lukasmueller** [comment]
I think the initial check should also be reported by email, as the check can take longer than the actual upload and time out.

**Review by lukasmueller — COMMENTED** [review]
The check results should also go through email? The check can take a long time and time out

**Review by lukasmueller — CHANGES_REQUESTED** [review]
The email could cotnain a link to each accession uploaded (maybe with a cutoff if there are more than a thousand or so). What do you think?

**Comment by dwaring87** [comment]
One minor issue: If I choose cancel on the confirmation dialog asking if I want to continue and receive an email, the accessions are still saved and I still get an email.  I doubt anyone would get that far and cancel at that point, but if I did, I would expect it not to proceed with saving.

**Review by dwaring87 — APPROVED** [review]

**Review by lukasmueller — APPROVED** [review]
:-)

**Pull request merged**